standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5067,6 +5069,25 @@ private void timeOutLocks(Connection dbConn) {
    */
   @RetrySemantics.Idempotent
   public void performTimeOuts() {
+    try {
+      // time out non-replication txns
+      String epochFn = getEpochFn(dbProduct);
+      String timeoutTxnsQuery = " \"TXN_ID\" FROM \"TXNS\" WHERE \"TXN_STATE\" = " + TxnStatus.OPEN +
+              " AND \"TXN_LAST_HEARTBEAT\" <  " + epochFn + "-" + timeout +
+              " AND \"TXN_TYPE\" != " + TxnType.REPL_CREATED.getValue();
+      performTimeOutsInternal(timeoutTxnsQuery);
```

Comment:
Why do we need 2 db connections here to perform the timeout? Could we change this to a single query to avoid doing FULL table scan several times?

Reply:
I did in the newest commit. However there's a mistake somewhere so maybe don't check yet :)

Reply:
Ok now it should be ok